.circleci/config.yml

```diff
@@ -541,19 +541,27 @@ commands:
           name: Split tests
           working_directory: ~/project/e2e_tests
           command: |
-            # If a test mark is specified, preselect only files that contain tests with that mark to
-            # minimize how wrong CircleCI's splitting can get things.
-            if [ -n "<<parameters.mark>>" ]; then
-              mark="<<parameters.mark>>"
-              # Extract the first mark for complex conditions such as 'mark1 and not mark2'.
-              mark_prefix=${mark%% and*}
-              find tests -name 'test*.py' -print0 | xargs -0 grep -rl "pytest\.mark\.$mark_prefix"
-            else
-              circleci tests glob 'tests/**/test*.py'
-            fi | circleci tests split --split-by=timings > /tmp/tests-to-run
+            # Preselect the files that contain any matching tests to minimize how wrong CircleCI's
+            # splitting can get things. (The presence of files with no tests can cause the tests
+            # with files to be bunched up incorrectly, making the split suboptimal.)
```

Comment:
Is it "the presence of files with no tests" or "tests with no files"?  The reversal of those terms in the second clause makes me wonder if I'm misunderstanding the explanation.

Reply:
Oops, "tests with files" should be "files with tests".